src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -328,18 +329,19 @@ abstract class UnCurry extends InfoTransform
             }
           }
         }
-        args.take(formals.length - 1) :+ (suffix setType formals.last)
+        args.take(params.length - 1) :+ (suffix setType params.last.info)
```

Comment:
By my reckoning, the `:+` method of the immutable List class, since it alters the list ending, needs to copy the whole List. Perhaps a `ListBuffer` would be better?

Reply:
Good idea, I've switch to a `ListBuffer`.